### PR TITLE
[Content] Documentation guidelines

### PR DESCRIPTION
**Pull request by ronijaakkola**
## Description
This PR adds documentation guidelines and basic instructions to site package README. The easiest way to review these changes is to browse the README directly in GitHub: https://github.com/City-of-Helsinki/helsinki-design-system/tree/content/documentation-guidelines-and-readme/site

Apart from the README, only very minor changes were done to the Contribution page (mainly linking to the new README).

## How Has This Been Tested?
This has been tested by running the documentation site locally. README has been checked in GitHub directly.

### CODE REVIEW

File: site/README.md [hunk 1]
```diff
@@ -1,70 +1,106 @@
 # Helsinki Design System documentation site
 
-Built with [docz](https://www.docz.site/).
+This is the public documentation site of Helsinki Design System ([hds.hel.fi](https://hds.hel.fi/)). It is built with [docz](https://www.docz.site/) which is powered by [Gatsby](https://www.gatsbyjs.com/). Documentation site includes:
+
```
Comment (eemelinieminen) [inline site/README.md:4]: pitäiskö lisätä listaan myös "General introduction of HDS"

File: site/README.md [hunk 2]
```diff
@@ -1,70 +1,106 @@
...
+- Design token documentation
+- Component documentation
+- About section with news, release notes and road map
+- Resources section
+- Contribution guidelines
 
 ## Getting started
 
-### Run the dev env
+When making changes to the documentation website, recommended way to get it running locally. This way you can review and see your changes in real time like they would show up in production.
```
Comment (eemelinieminen) [inline site/README.md:16]: "recommended way to get it running locally" lauseeseen on jääny joku vähän oudosti

**Review by eemelinieminen — COMMENTED** [review]
Really clear guidelines. Found only a couple of minor fixes.

File: site/README.md (hunk 2 above)
Comment (ronijaakkola) [reply site/README.md]: Korjattu.

File: site/README.md (hunk 1 above)
Comment (ronijaakkola) [reply site/README.md:4]: Korjattu.

File: site/README.md [hunk 3]
```diff
@@ -1,70 +1,111 @@
...
+Make a change to any of the documentation `.mdx` files and refresh the page. You can also try to clear browser's cache or doing a hard refresh.
 
-    ```
-    TypeError: Cannot read property 'find' of undefined
-    ```
-  
-    Make a change to any of the documentation `.mdx` files and refresh the page.
+**After changes the page renders completely empty or some of the content does not show on the page**
+
+If you have used custom React components inlined (like the Link component), Gatsby does not render those paragraphs correctly. The fix is to wrap the text into paragraph tags. This way Gatsby
```
Comment (eemelinieminen) [inline site/README.md:111]: Vika lause katkeaa kesken. Onko lopusta pudonnut jotain pois?

File: site/docs/contributing.mdx [hunk 4]
```diff
@@ -82,18 +82,12 @@ _Coming soon!_
...
+<p>When contributing for documentation, it is recommended to set up a local development environment. This allows you to view changes you make in real time. Refer to <Link href="https://github.com/City-of-Helsinki/helsinki-design-system/tree/master/site#setting-up-local-development-environment" external>site project GitHub README</Link> for easy instructions how to set up the local development envinroment.</p>
 
 ### Guidelines for writing documentation
 
-_Coming soon!_
+<p>After the local deveopment environment is set up, editing documentation and adding new pages is simple. Refer to <Link href="https://github.com/City-of-Helsinki/helsinki-design-system/tree/master/site#writing-documentation" external>site project GitHub README</Link> for general documentation guidelines and instructions on how to create new pages and categories.</p>
```
Comment (eemelinieminen) [inline site/docs/contributing.mdx:91]: deveopment > development

File: site/docs/contributing.mdx [hunk 5]
```diff
@@ -82,18 +82,12 @@ _Coming soon!_
...
-
-The easiest way to edit a page is with the Edit this page on Github link that appears at the bottom of every page on the site. The link opens the specific Github page where you can edit the content and propose the change with a pull request.
-
-### Forking repository and setting up local environment
-
-<p>If you are planning regular or more comprehensive content updates, you will want to fork the repository and install some of the tools we use to build the website. This will create an easier workflow for you long term. Follow README in <Link href="https://github.com/City-of-Helsinki/helsinki-design-system/tree/master/site" external>documentation site package</Link> to get you started setting up local development environment.</p>
+<p>When contributing for documentation, it is recommended to set up a local development environment. This allows you to view changes you make in real time. Refer to <Link href="https://github.com/City-of-Helsinki/helsinki-design-system/tree/master/site#setting-up-local-development-environment" external>site project GitHub README</Link> for easy instructions how to set up the local development envinroment.</p>
```
Comment (eemelinieminen) [inline site/docs/contributing.mdx:87]: vika sana: envinroment > environment

File: site/README.md (hunk 3 above)
Comment (ronijaakkola) [reply site/README.md:111]: Korjattu.

File: site/docs/contributing.mdx (hunk 4 above)
Comment (ronijaakkola) [reply site/docs/contributing.mdx:91]: Korjattu.

File: site/docs/contributing.mdx (hunk 5 above)
Comment (ronijaakkola) [reply site/docs/contributing.mdx:87]: Korjattu.

**Review by eemelinieminen — APPROVED** [review]

**Pull request merged**